xtable-api/src/main/java/org/apache/xtable/model/catalog/CatalogTableIdentifier.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+ 
+package org.apache.xtable.model.catalog;
+
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Value;
+
+/** This class represents the unique identifier for a table in a catalog. */
+@Value
+@Builder
+public class CatalogTableIdentifier {
```

Comment:
1) What do you think of calling this class `TableIdentifier`. 
2) Also, we should consider making the naming more generic, since it should represent all types of table identifiers. For instance, while `databaseName` is a popular namespace, there's also `schema`. In some scenarios, `databaseName` is synonymous with `catalogName`. The current two-part naming based on `table` and `databaseName` seems a bit restrictive to me.

Would you mind sharing the use cases this naming caters to?

Reply:
1 is okay, included the prefix Catalog because `TableIdentifier` is overloaded in the dependent projects (iceberg, hudi, delta etc.) and didn't want to add another identifier with the same name. 

For 2, agreed that each catalog or system has a different term for what's called a "logical grouping of tables". 
I have looked up this name in different systems and have found databaseName, namespace, schema are the popular ones, a more generic name that comes to my mind is `tableCollection` or `tableGroup` ? If we can't find a generic name, choosing databaseName maybe okay ? Regardless of the name we choose the conversion interfaces for each catalog are responsible for translating the catalog table definition to this representation.